pkg/cmd/node/node_list_components.go

```diff
@@ -0,0 +1,98 @@
+package node
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/spf13/cobra"
+)
+
+func newCmdNodeListComponents(ctx api.Context) *cobra.Command {
+	var jsonOutput bool
+	var leader bool
+	var mesosID string
+	cmd := &cobra.Command{
+		Use:   "list-components",
+		Short: "Print a list of available DC/OS components on specified node.",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !leader && mesosID == "" {
+				return fmt.Errorf("'--leader' or '--mesos-id' must be provided")
+			} else if leader && mesosID != "" {
+				return fmt.Errorf("unable to use leader and mesos id at the same time")
+			}
+
+			client := mesosClient()
+			ip := ""
+			if leader {
+				masters, err := client.Leaders()
+				if err != nil {
+					return err
+				}
+
+				if len(masters) != 1 {
+					return fmt.Errorf("expecting one leader. Got %d", len(masters))
+				}
+
+				if masters[0].IP == "" {
+					return fmt.Errorf("invalid leader response, missing field 'ip'")
+				}
+				ip = masters[0].IP
+			} else {
+				agents, err := client.Agents()
+				if err != nil {
+					return err
+				}
+				for _, agent := range agents {
+					if mesosID == agent.AgentInfo.GetHostname() {
+						ip = agent.AgentInfo.GetHostname()
+					}
+				}
+				if ip == "" {
+					return fmt.Errorf("agent '%s' not found", mesosID)
+				}
+			}
+
+			// Reuse the existing HTTP client of the Mesos client instead of creating a new one.
+			resp, err := client.HTTP().Get(fmt.Sprintf("/system/health/v1/nodes/%s/units", ip))
```

Comment:
Is this the only place in the CLI a call to this API is made? I also don't think it belongs in the `mesos` package, but having it in the `RunE()` func of the command while still using the mesos client seems not clean to me. This and the respective type should go in a separate package (`health` maybe?) IMO. Will make the `RunE()` more readable and shorter.